src/fairseq2/datasets/asr.py

```diff
@@ -344,4 +350,177 @@ def splits(self) -> set[str]:
         return self._splits
 
 
+class HuggingFaceAsrDataset(AsrDataset):
+    _name: str
+    _hg_dataset: datasets.dataset_dict.DatasetDict
+    _splits: set[str]
+
+    def __init__(
+        self,
+        name: str,
+        hg_hub_name: str = "mozilla-foundation/common_voice_17_0",
+        hg_hub_data_dir: str = "ia",
+        splits: set[str] = {"train", "validation", "test"},
+    ) -> None:
+        self._name = name
+        self._splits = splits
+        print("Loading dataset from HuggingFace Hub...")
```

Comment:
Instead of `print()`, I suggest using `log.info()` to play nicely with the trainer output.